app-editors/emacs/emacs-26.1.recipe

```diff
@@ -0,0 +1,135 @@
+SUMMARY="The famous text editor"
+DESCRIPTION="An extensible, customizable, free/libre text editor — and more.
+At its core is an interpreter for Emacs Lisp, a dialect of the Lisp \
+programming language with extensions to support text editing.
+
+* Content-aware editing modes, including syntax coloring, for many file types.
+* Complete built-in documentation, including a tutorial for new users.
+* Full Unicode support for nearly all human scripts.
+* Highly customizable, using Emacs Lisp code or a graphical interface.
+* An entire ecosystem of functionality beyond text editing, including a \
+project planner, mail and news reader, debugger interface, calendar, and more.
+* A packaging system for downloading and installing extensions."
+HOMEPAGE="https://gnu.org/s/emacs/"
+COPYRIGHT="2001-2018 Free Software Foundation, Inc."
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="http://www.artfiles.org/gnu.org/emacs/emacs-$portVersion.tar.xz"
+CHECKSUM_SHA256="1cf4fc240cd77c25309d15e18593789c8dbfba5c2b44d8f77c886542300fd32c"
+PATCHES="emacs-$portVersion.patchset"
+ADDITIONAL_FILES="emacs.rdef.in"
+
+ARCHITECTURES="?x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="?x86"
+
+# On x86_gcc2 we don't want to install the commands in bin/<arch>/, but in bin/.
+commandSuffix=$secondaryArchSuffix
+commandBinDir=$binDir
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	commandSuffix=
+	commandBinDir=$prefix/bin
+fi
+
+PROVIDES="
+	emacs$secondaryArchSuffix = $portVersion
+	cmd:ctags.emacs$commandSuffix = $portVersion
+	cmd:ebrowse$commandSuffix = $portVersion
+	cmd:emacs$commandSuffix = $portVersion
+	cmd:emacs_$portVersion$commandSuffix = $portVersion
+	cmd:emacsclient$commandSuffix = $portVersion
+	cmd:etags$commandSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libgnutls$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libncurses$secondaryArchSuffix
+	lib:libxml2$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libgnutls$secondaryArchSuffix
+	devel:libiconv$secondaryArchSuffix
+	devel:libncurses$secondaryArchSuffix
+	devel:libxml2$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:autoconf
+	cmd:aclocal
+	cmd:find
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	cmd:makeinfo
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:tar
+	cmd:which
+	"
+
+defineDebugInfoPackage emacs$secondaryArchSuffix \
+	$commandBinDir/ctags.emacs \
+	$commandBinDir/emacs-$portVersion \
+	$commandBinDir/emacsclient \
+	$commandBinDir/etags
+
+PATCH()
+{
+	# Hard links not supported on BFS.
+	sed -i 's/ln -f/ln -s/g' src/Makefile.in
+}
+
+BUILD()
+{
+	export LDFLAGS="-lnetwork -lbsd"
+	CFLAGS="-D_BSD_SOURCE"
+	export HOME=/boot/home
```

Comment:
This should be contained inside the build folder. We don't want it to mess with user's home when they build the recipe without chroot.

Also:

```diff
-	export LDFLAGS="-lnetwork -lbsd"
-	CFLAGS="-D_BSD_SOURCE"
+	export LIBS="-Wl,--as-needed -lnetwork -lbsd"
+	export CPPFLAGS="-D_BSD_SOURCE"
```

`-Wl,--as-needed` will avoid linking those libraries if they are unneeded.